### PR TITLE
Resolves HELIO-3613 modify auth API

### PR DESCRIPTION
**Pull request by gkostin1966**
Added product access API

      #
      # Access { :full, :read, :none }
      #
      #   :full - both read and download access
      #   :read - read access (no download access)
      #   :none - no access, but has subscription
      #
      #   :undefined - no subscription
      #

      test access -> def product_individual_access?(product_identifier:, individual_identifier:, access_type:)
      get access  -> def product_individual_access(product_identifier:, individual_identifier:)
      set access  -> def access_product_individual(product_identifier:, individual_identifier:, access_type:)

      test access -> def product_institution_access?(product_identifier:, institution_identifier:, access_type:)
      get access  -> def product_institution_access(product_identifier:, institution_identifier:)
      set access  -> def access_product_institution(product_identifier:, institution_identifier:, access_type:)

### CODE REVIEW

File: lib/turnsole/heliotrope/service.rb [hunk 1]
```diff
@@ -290,6 +306,78 @@ def unsubscribe_product_institution(product_identifier:, institution_identifier:
...
+        id = find_individual(identifier: individual_identifier)
+        return :undefined if product_id.nil? || id.nil?
+
+        response = connection.get("products/#{product_id}/individuals/#{id}/access")
+        return :undefined unless response.success?
+
+        response.body["access"].to_sym
+      end
+
+      def access_product_individual(product_identifier:, individual_identifier:, access_type:)
```
Comment (jgmorse) [inline lib/turnsole/heliotrope/service.rb:336]: I don't understand the difference between this method and [subscribe_product_individual](https://github.com/mlibrary/turnsole/pull/14/files#diff-b0b53bcfa5d13675f7277c74c07a8fc8b794306004be1c6354ea1d2d6a528cc0R248). 

Is the difference that this method is _changing_ an access type on an existing subscription? Whereas the other method is creating a new subscription and setting its access type?

I think their names need to change to something more illustrative, to make the distinction clearer.

File: lib/turnsole/heliotrope/service.rb [hunk 2]
```diff
@@ -290,6 +306,78 @@ def unsubscribe_product_institution(product_identifier:, institution_identifier:
...
       end
 
+      #
+      # Access { :full, :read, :none }
+      #
+      #   :full - both read and download access
+      #   :read - read access (no download access)
+      #   :none - no access, but has subscription
+      #
+      #   :undefined - no subscription
```
Comment (jgmorse) [inline lib/turnsole/heliotrope/service.rb:316]:
```suggestion
      #   :undefined - no association between subscriber and product
```

File: lib/turnsole/heliotrope/service.rb [hunk 3]
```diff
@@ -290,6 +306,78 @@ def unsubscribe_product_institution(product_identifier:, institution_identifier:
         response.success?
       end
 
+      #
+      # Access { :full, :read, :none }
+      #
+      #   :full - both read and download access
+      #   :read - read access (no download access)
+      #   :none - no access, but has subscription
```
Comment (jgmorse) [inline lib/turnsole/heliotrope/service.rb:314]:
```suggestion
      #   :none - expired access
```

File: lib/turnsole/heliotrope/service.rb [hunk 4]
```diff
@@ -290,6 +306,78 @@ def unsubscribe_product_institution(product_identifier:, institution_identifier:
...
+        id = find_institution(identifier: institution_identifier)
+        return :undefined if product_id.nil? || id.nil?
+
+        response = connection.get("products/#{product_id}/institutions/#{id}/access")
+        return :undefined unless response.success?
+
+        response.body["access"].to_sym
+      end
+
+      def access_product_institution(product_identifier:, institution_identifier:, access_type:)
```
Comment (jgmorse) [inline lib/turnsole/heliotrope/service.rb:367]: I don't understand the difference between this method and [subscribe_product_institution](https://github.com/mlibrary/turnsole/pull/14/files#diff-b0b53bcfa5d13675f7277c74c07a8fc8b794306004be1c6354ea1d2d6a528cc0R283)

Is the difference that this method is changing an access type on an existing subscription? Whereas the other method is creating a new subscription and setting its access type?

I think their names need to change to something more illustrative, to make the distinction clearer.

**Review by jgmorse — CHANGES_REQUESTED** [review]

**Comment by sethaj** [comment]
Closing this. The changes will be in https://github.com/mlibrary/turnsole/pull/15  instead.

**Pull request closed without merging**